qt/applications/workbench/workbench/config/user.py

```diff
@@ -82,25 +71,14 @@ def get(self, option, second=None, type=None):
         ``config.get('main/window/size')`` If no option is found then
         a KeyError is raised
         """
-
-        def raise_keyerror(key):
-            raise KeyError('Unknown config item requested: "{}"'.format(option))
-
         full_option = self._check_section_option_is_valid(option, second)
         if type is None:
-            # return whatever the QSettings object returns
-            value = self.qsettings.value(full_option)
-            if value is None:
-                raise_keyerror(full_option)
-            else:
-                return value
-        else:
-            # PyQt QSettings with the type parameter tries to always return the type even if no value exists
-            # We still want a KeyError if it does not exist
-            if self.has(option, second):
-                return self.qsettings.value(full_option, type=type)
-            else:
-                raise_keyerror(option)
+            # Some platforms only ever store string values in QSettings so type of stored settings can get lost
+            raise RuntimeError("Please specify the type of the setting you are retrieving.")
```

Comment:
Can we raise a[`TypeError`](https://docs.python.org/3/library/exceptions.html#TypeError) here?

Reply:
I've changed it to be a `ValueError` instead, because the try catch statement in the comment above will catch `TypeError`'s, and I think we don't want this error to be caught as it is a warning to the programmer